File: mathenger/Mathenger/Mathenger/services/AccountService.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using Mathenger.config;
using Mathenger.models;
using RestSharp;

namespace Mathenger.services
{
    public class AccountService
    {
        private RequestSender _sender;

        public AccountService(RequestSender sender)
        {
            _sender = sender;
        }

        public void GetCurrentAccount(Action<Account> accountConsumer)
        {
            var request = new RestRequest("/account/me");
            _sender.Send(request, accountConsumer);
        }

        public void GetMyContacts(Action<ObservableCollection<Account>> contactsConsumer)
        {
            var request = new RestRequest("/account/me/contacts");
            _sender.Send(request, contactsConsumer);
        }

        public void Search(string search, Action<ObservableCollection<Account>> accountsConsumer)
        {
            var request = new RestRequest("/account/search");
            request.AddParameter("search", search);
            _sender.Send(request, accountsConsumer);
        }

        public void AddContact(long id, Action<Account> contactConsumer)
        {
            var request = new RestRequest($"/account/me/contacts/new/{id}", Method.POST);
            _sender.Send(request, contactConsumer);
        }

        public void DeleteContact(long id, Action onSuccess)
        {
            var request = new RestRequest($"/account/me/contacts/delete/{id}", Method.DELETE);
            _sender.Send(request, onSuccess);
        }
    }
}

File: mathenger-ui/src/services/account.service.js
import axios from 'axios';

export const accountService = {
    getCurrentAccount,
    getMyContacts,
    addContact,
    deleteContact,
    search
}

function getCurrentAccount() {
    return axios.get("/account/me")
        .then(response => response.data);
}

function getMyContacts() {
    return axios.get("/account/me/contacts")
        .then(response => response.data);
}

function addContact(contactId) {
    return axios.post(`/account/me/contacts/${contactId}`)
        .then(response => response.data);
}

function deleteContact(contactId) {
    return axios.delete(`/account/me/contacts/${contactId}`);
}

function search(str) {
    return axios.get("/account/search", {
        params: {
            search: str
        }
    }).then(response => response.data);
}
